Python SDK doc revamp (#1165)
rudderlabs/rudderstack-docs: docs/sources/event-streams/sdks/rudderstack-python-sdk.mdx

description: Use RudderStack’s Python SDK to send events to various destinations.
---

RudderStack's Python SDK lets you track and send the events from your Python applications to the specified destinations.

> <div class="infoBlock">

Reviewer: can remove this infoblock for consistency purposes

> <div class="infoBlock">

Author: ```suggestion
```

Refer to the SDK's <a href="https://github.com/rudderlabs/rudder-sdk-python">GitHub codebase</a> for the implementation-specific details.

> </div>

Author: ```suggestion
```

> </div>

Author: ```suggestion
```

## SDK setup requirements

1. Sign up to [RudderStack Cloud](https://app.rudderstack.com/signup).
2. <Link to="/dashboard-guides/sources/#adding-a-source">Set up a Python source</Link> in your dashboard. You should be able to see a <strong>write key</strong> for this source.

> You will also need a data plane URL. Refer to the <Link to="/dashboard-guides/overview/#data-plane-url">Dashboard Overview</Link> guide for more information on the data plane URL and where to find it.

Reviewer: can move this after the image as write key reflects in the image.

Author: as the webapp is not yet updated with the latest snippet, removed the image temporarily - will add the updated image later.

> <img src="../../../assets/event-stream-sources/python-source-write-key.png" alt="Python source write key" />

Author: ```suggestion
```

> To initialize the RudderStack client, run the following code snippet:

Reviewer: ```suggestion
To initialize the SDK, run the following code snippet:
```

same in the heading:
Initializing the RudderStack client -> Initializing the SDK

> To initialize the RudderStack client, run the following code snippet:

Author: ```suggestion
To initialize the SDK, run the following code snippet:
```

> The `page` method parameters are as described below:

Reviewer: ```suggestion
The `page` method parameters are as shown:
```

> The `screen` method parameters are as described below:

Reviewer: ```suggestion
The `screen` method parameters are as shown:
```

| `properties` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>  | Object | An optional dictionary of the properties associated with the event.   |
| `context`      | Object   |  An optional dictionary of information that provides context about the event. It is not directly related to the API call.      |
| `integrations` | Object | An optional dictionary containing the destinations to be either enabled or disabled. |
| `timestamp`    | Timestamp in ISO 8601 format     | The timestamp of the event's arrival.                |
| `message_id` | String | A unique identifier for the event. The SDK generates this field automatically. It is **highly recommended** to not send this field unless there is a specific requirement or use case. | 

### Group

The <Link to="/event-spec/standard-events/group/">`group`</Link> call lets you link an identified user with a group, such as a company, organization, or an account. It also lets you record any custom traits or properties associated with that group.

A sample `group` call made using the Python SDK is shown below:

```python
rudder_analytics.group('1hKOmRA4GRlm', '12', {
    'name': 'Company',
    'domain': 'IT'
})
```

> The `group` method parameters are as follows:

Reviewer: ```suggestion
The `group` method parameters are as shown:
```

| **Field**      | **Type** |  **Description**     |
| :------------- | :------- |  :---------------- |
| `user_id` <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>anonymous_id</strong> is absent.</span>     | String   | Unique identifier for a user in your database.      |
| `anonymous_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required, if <strong>user_id</strong> is absent.</span> | String   |  The SDK automatically sets this identifier in cases where there is no unique identifier for the user. |
| `group_id`  <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>    | String   | Unique identifier of the group in your database.    |

> | `traits`       | Object   | An optional dictionary of the group's traits like `name`or `email`.     |

Reviewer: ```suggestion
| `traits`       | Object   | An optional dictionary of the group's traits like `name` or `email`.     |
```

| `context`      | Object   |  An optional dictionary of information that provides context about the event. It is not directly related to the API call.      |
| `timestamp`    | Timestamp in ISO 8601 format     | The timestamp of the event's arrival.                |
| `message_id` | String | A unique identifier for the event. The SDK generates this field automatically. It is **highly recommended** to not send this field unless there is a specific requirement or use case. | 

### Alias

The <Link to="/event-spec/standard-events/alias/">`alias`</Link> call lets you merge different identities of a known user. It is an advanced method that lets you change the tracked user's ID explicitly. You can use `alias` for managing the user's identity in some of the downstream destinations.

<div class="warningBlock">
RudderStack supports sending <code class="inline-code">alias</code> events only to select downstream destinations. Refer to the <Link to="/destinations/streaming-destinations/">destination-specific documentation</Link> for more details.
</div>

A sample `alias` call is as shown:

```python
rudder_analytics.alias('previous_id', 'user_id')
```

> The `alias` method parameters are as mentioned below:

Reviewer: ```suggestion
The `alias` method parameters are as shown:
```
